component/authenticator/src/webapp/voiceMobile.jsp

```diff
@@ -0,0 +1,203 @@
+<%@ page import="org.wso2.carbon.identity.application.authentication.endpoint.util.Constants" %>
+<%@ page import="java.io.File" %>
+<%@ page import="java.util.Map" %>
+<%@ page import="org.apache.commons.lang.StringUtils"%>
+<%@ page language="java" contentType="text/html; charset=UTF-8" pageEncoding="UTF-8" %>
+<%@ include file="includes/localize.jsp" %>
+<%@ page import="static java.util.Base64.getDecoder" %>
+<%@ page import="org.wso2.carbon.identity.authenticator.voiceotp.VoiceOTPConstants" %>
+<%@ page import="org.apache.commons.lang.StringUtils" %>
+<%@ taglib prefix="layout" uri="org.wso2.identity.apps.taglibs.layout.controller" %>
+
+<jsp:directive.include file="includes/layout-resolver.jsp"/>
+
+<%
+    request.getSession().invalidate();
+    String queryString = request.getQueryString();
+    Map<String, String> idpAuthenticatorMapping = null;
+    String mobileRegex = null;
+    boolean validateMobileNumberFormat = false;
+    String mobileRegexPolicyValidationErrorMessage = null;
+    if (request.getAttribute(Constants.IDP_AUTHENTICATOR_MAP) != null) {
+        idpAuthenticatorMapping = (Map<String, String>) request.getAttribute(Constants.IDP_AUTHENTICATOR_MAP);
+    }
+    if (StringUtils.isNotBlank(request.getParameter(VoiceOTPConstants.MOBILE_NUMBER_REGEX_PATTERN))) {
+        mobileRegex = new String(getDecoder().decode(request.getParameter(VoiceOTPConstants.MOBILE_NUMBER_REGEX_PATTERN)));
+        validateMobileNumberFormat = true;
+    }
+    if (StringUtils.isNotBlank(request.getParameter(VoiceOTPConstants.MOBILE_NUMBER_PATTERN_POLICY_FAILURE_ERROR_MESSAGE))) {
+        mobileRegexPolicyValidationErrorMessage = new String(getDecoder().decode(request.getParameter(VoiceOTPConstants.MOBILE_NUMBER_PATTERN_POLICY_FAILURE_ERROR_MESSAGE)));
+    }
+
+    String errorMessage = IdentityManagementEndpointUtil.i18n(resourceBundle,"error.retry");
+    boolean authenticationFailed = false;
+
+    if (Boolean.parseBoolean(request.getParameter(Constants.AUTH_FAILURE))) {
+        authenticationFailed = true;
+
+        if (request.getParameter(Constants.AUTH_FAILURE_MSG) != null) {
+            errorMessage = request.getParameter(Constants.AUTH_FAILURE_MSG);
+
+                if (errorMessage.equalsIgnoreCase("authentication.fail.message")) {
+                    errorMessage = "Authentication Failed! Please Retry";
+                }
+
+                if (StringUtils.isNotBlank(request.getParameter("authFailureInfo"))) {
+                    errorMessage = request.getParameter("authFailureInfo");
+                }
+        }
+    }
+%>
+
+<%-- Data for the layout from the page --%>
+<%
+    layoutData.put("containerSize", "medium");
+%>
+
+<html lang="en-US">
+    <head>
+        <%-- header --%>
+        <%
+            File headerFile = new File(getServletContext().getRealPath("extensions/header.jsp"));
+            if (headerFile.exists()) {
+        %>
+        <jsp:include page="extensions/header.jsp"/>
```

Comment:
Why importing `header.jsp` from extensions is here? he extensions is to do UI customisation (CSS changes) to the webapps. As this is a product feature such extensions are not required.